api/subjects/serializers.py

```diff
@@ -0,0 +1,117 @@
+from rest_framework import serializers as ser
+from rest_framework import exceptions
+
+from framework.auth.core import Auth
+from framework.exceptions import PermissionsError
+
+from api.base.serializers import (
+    JSONAPISerializer,
+    JSONAPIRelationshipSerializer,
+    LinksField,
+    RelationshipField,
+    BaseAPISerializer,
+)
+from osf.exceptions import NodeStateError, ValidationValueError
+
+
+class UpdateSubjectsMixin(object):
+    def update_subjects_method(self, resource, subjects, auth):
+        # Method to update subjects on resource
+        raise NotImplementedError()
+
+    def update_subjects(self, resource, subjects, auth):
+        """Updates subjects on resource and handles errors.
+
+        :param object resource: Object for which you want to update subjects
+        :param list subjects: Subjects array (or array of arrays)
+        :param object Auth object
+        """
+        try:
+            self.update_subjects_method(resource, subjects, auth)
+        except PermissionsError as e:
+            raise exceptions.PermissionDenied(detail=str(e))
+        except ValueError as e:
+            raise exceptions.ValidationError(detail=str(e))
+        except ValidationValueError as e:
+            raise exceptions.ValidationError(detail=e[0])
+        except NodeStateError as e:
+            raise exceptions.ValidationError(detail=str(e))
+
+
+class SubjectSerializer(JSONAPISerializer):
+    filterable_fields = frozenset([
+        'text',
+        'parent',
+        'id',
+    ])
+    id = ser.CharField(source='_id', required=True)
+    text = ser.CharField(max_length=200)
+    share_title = ser.CharField(source='provider.share_title', read_only=True)
```

Comment:
Since this serializer doesn't need to worry about back compat, can we take this opportunity to rename `share_title` to something clearer? Since it refers to the name of the taxonomy the subject belongs to, maybe `taxonomy_name`, or `provider_taxonomy_name`?

Reply:
name of taxonomy in which the subject lives, like bepress.  Taxonomy object - bepress taxonomy, provider has a relationship to and subject has a relationship to.